provider/cluster/cluster_resource.go

```diff
@@ -646,6 +692,33 @@ func populateClusterState(object *cmv1.Cluster, state *ClusterState) error {
 		state.AWSSubnetIDs = awsSubnetIds
 	}
 
+	additionalComputeSecurityGroupIds, ok := object.AWS().GetAdditionalComputeSecurityGroupIds()
+	if ok {
+		awsAdditionalSecurityGroupIds, err := common.StringArrayToList(additionalComputeSecurityGroupIds)
+		if err != nil {
+			return err
+		}
+		state.AWSAdditionalComputeSecurityGroupIds = awsAdditionalSecurityGroupIds
+	}
```

Comment:
nit: I think you can use one liner for the if in all places like this

```suggestion
	if additionalComputeSecurityGroupIds, ok := object.AWS().GetAdditionalComputeSecurityGroupIds(); ok {
		awsAdditionalSecurityGroupIds, err := common.StringArrayToList(additionalComputeSecurityGroupIds)
		if err != nil {
			return err
		}
		state.AWSAdditionalComputeSecurityGroupIds = awsAdditionalSecurityGroupIds
	}
```

Reply:
So, I do agree with this, but all code around the changes are also not using the one liner version. So I'm thinking of keeping this for consistency and we can update all at once as a refactor. What do you think?